src/engine/Engine.cpp

```diff
@@ -1821,6 +1824,27 @@ List<unsigned> Engine::getInputVariables() const
     return _preprocessedQuery.getInputVariables();
 }
 
+void Engine::performSimulation()
+{
+    // outer vector is for neuron
+    // inner vector is for simulation value
+    std::vector<std::vector<double>> simulations;
+
+    std::mt19937 mt( GlobalConfiguration::SIMULATION_RANDOM_SEED );
+
+    for ( unsigned i = 0; i < _networkLevelReasoner->getLayer( 0 )->getSize(); ++i )
+    {
+        std::uniform_real_distribution<double> distribution( _networkLevelReasoner->getLayer( 0 )->getLb( i ),
+                                                                _networkLevelReasoner->getLayer( 0 )->getUb( i ) ); 
+        std::vector<double> simulationInput( Options::get()->getInt( Options::NUMBER_OF_SIMULATIONS ) );
```

Comment:
Store NUMBER_OF_SOMULATIONS as a field variable in the constructor. Since parallel mode can call processInputQuery() in parallel, and the Options object is not thread safe.

Reply:
done.